### PR TITLE
INTYGFV-15612: DatePicker, Calendar & Slutdatum

### PR DESCRIPTION
**Pull request by SamuelTilly**

### CODE REVIEW

**Review by ChrilleSigerhed — CHANGES_REQUESTED** [review]
Awsome, you really done a great job with this component! Very impressive.

The component works great, the only functional flaw that I could find is that when clicking the reset filter-button the values does not reset.

Code-wise I suggest we add some tests for the classes that is missing tests. If you feel like it's redundant for some of the components I can understand that. 

For me the tests also can work as documentation of the components, so I think it could make the components easier to understand aswell!

Again, great work!

File: apps/rehabstod/src/utils/TestProvider.tsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
 import { ReactNode } from 'react'
+import { I18nProvider } from 'react-aria'
 import { Provider } from 'react-redux'
 import { MemoryRouter } from 'react-router-dom'
 import { store } from '../store/store'
 
+// createMemoryRouter()
```
Comment (ChrilleSigerhed) [inline apps/rehabstod/src/utils/TestProvider.tsx:7]: Remove this comment?

File: apps/rehabstod/src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const context = useContext(PopoverContext)
+
+  if (context == null) {
+    throw new Error('Popover components must be wrapped in <Popover />')
+  }
+
+  return context
+}
+
+export function Popover({ children, ...props }: PopoverProps & { children: ReactNode }) {
```
[the remark this answers is not in the document]
Comment (SamuelTilly) [reply apps/rehabstod/src/components/Popover/Popover.tsx:71]: Not sure what tests I could add here, most of the logic will only affect styling and that is not something we test for usually. Elements will for example always be shown, only hidden with positioning and height styling.

[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply apps/rehabstod/src/components/Popover/Popover.tsx:71]: Makes sense.

**Review by ChrilleSigerhed — APPROVED** [review]
Awsome, really impressive! Remember to fix the userEvent setup before merging.

**Pull request merged**